src/main/java/com/evilcorp/bloom/service/CategoryService.java

```diff
@@ -0,0 +1,75 @@
+package com.evilcorp.bloom.service;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.dao.DataIntegrityViolationException;
+
+import com.evilcorp.bloom.dto.CategoryDto;
+import com.evilcorp.bloom.model.Category;
+import com.evilcorp.bloom.repo.CategoryRepo;
+import com.evilcorp.bloom.exception.NestedSubcategoriesException;
+import com.evilcorp.bloom.exception.NotFoundException;
+
+public class CategoryService {
+
+  private final CategoryRepo categoryRepo;
+
+  public CategoryService(CategoryRepo categoryRepo) {
+    this.categoryRepo = categoryRepo;
+  }
+
+  public String capitalize(String categoryName) {
+    if (categoryName.isEmpty() || categoryName == null) {
+      return categoryName;
+    }
+
+    return Arrays.stream(categoryName
+        .split("\\s+"))
```

Comment:
We should create a constant for this magical string value so it is easier to read for people who are used to regex and people who are not, and we can re-use this string value in other calls to this function, avoiding having to create and destroy this string value all the time (memory and process impact is very very very low, but if it is still a benefit)

Reply:
done